src/windows/TextInput.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+* TextInput.tsx
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*
+* RN Windows-specific implementation of the cross-platform TextInput abstraction.
+*/
+
+import React = require('react');
+import RN = require('react-native');
+
+import {applyFocusableComponentMixin, FocusManagerFocusableComponent } from '../native-desktop/utils/FocusManager';
+
+import {TextInput as TextInputBase} from '../native-common/TextInput';
+
+export class TextInput extends TextInputBase implements FocusManagerFocusableComponent {
+
+    protected _render(props: RN.TextInputProps): JSX.Element {
+
+        return (
+            <RN.TextInput
+            {...props}
+            tabIndex = {this.getTabIndex()}
+            onFocus = { (e: React.FocusEvent<any>) => this._onFocusEx(e, props.onFocus)}
```

Comment:
>onFocus [](start = 12, length = 7)

Are you stomping on the super class's `onFocus` handler?

Reply:
I think I was trying to be fancy by not having assumptions on what the base class passed in the props. The arrow function though is created one each render now, kind of bad. I'll try to reshuffle.

---
In reply to: [159359055](https://github.com/Microsoft/reactxp/pull/404#discussion_r159359055) [](ancestors = 159359055)